root/etc/systemd/system/discovery-menu.service

```diff
@@ -1,7 +1,7 @@
 [Unit]
 Description=Display interactive TUI on tty1
 Wants=basic.target
-After=basic.target network.target network-online.target nss-lookup.target
+After=basic.target network.target nss-lookup.target nm-prepare.service
```

Comment:
On the other hand, menu should NOT require network online, because in PXE-less network is usually not configured at all in this workflow.